### DISCUSSION TITLE
MOOSE on WSL2/Ubuntu-20.04

### DISCUSSION THREAD

**Discussion by oMuransky** [idaholab/moose#22764] [Q&A General] [answered]
Hi Guys, 

I'm trying to repeat the MOOSE installation process from a few of weeks ago on WSL2/Ubuntu but it does not work anymore. When running "make" I'm getting a fatal error (missing file:  MooseRevision.h)

My current installation process on clean WSL2/Ubuntu: 
```
#---------------
# Update Ubuntu & Install pre-requisites 
#---------------
sudo apt update 
sudo apt upgrade
sudo apt install build-essential python3 libmpich-dev git python-is-python3 cmake m4 python3-dev 
sudo apt install liblapack-dev libblas-dev libssl-dev bison flex
cd ~
#---------------
# Install/Update cmake
#---------------
wget https://github.com/Kitware/CMake/releases/download/v3.23.0/cmake-3.23.0.tar.gz
tar -zxvf cmake-3.23.0.tar.gz 
cd cmake-3.23.0/
./bootstrap
make
sudo make install
#---------------
# Remove mpich & Install openmpi
#---------------
sudo apt-get purge --auto-remove mpich 
sudo apt install openmpi-bin
which runmpi
runmpi --version
#---------------
# Install MOOSE
#---------------
mkdir moose
cd moose
git clone https://github.com/idaholab/moose.git
cd moose/
cd scripts/
./update_and_rebuild_petsc.sh --download-mumps=no --with-mumps=no
./update_and_rebuild_libmesh.sh 
cd ../framework/
make -j 8 <== fails here (fatal error: MooseRevision.h: No such file or directory) 
```

![error](https://user-images.githubusercontent.com/28768162/202926103-1fb31242-6aee-4554-92ff-cb2d4ed8cbff.PNG)

**Comment by dschwen** [comment]
Can you scroll up and check if there are any other errors or warnings further up?

**Reply by oMuransky** [reply]
Hello Daniel, 

No errors or warnings during the "moose make" process, however, a few warnings during the "./update_and_rebuild_libmesh.sh" process (attached) - not sure why I'm getting these hdf5 warnings because it is installing the hdf5 while running "./update_and_rebuild_petsc.sh --download-mumps=no --with-mumps=no" 

![error](https://user-images.githubusercontent.com/28768162/202926817-a65b5b54-28c6-40fd-8548-af67ba0ceb14.PNG)

**Reply by GiudGiud** [reply]
this mooserevision.h missing error is a missing python package

**Reply by GiudGiud** [reply]
see this
https://github.com/idaholab/moose/discussions/22035#discussioncomment-3588934

**Comment by oMuransky** [comment]
Thanks for pointing me to this thread, Guillaume. Interesting, I installed "python3-dev", but the "packaging" was still missing (this is new). All what needs to be done is: "pip install packaging" and then the "moose make" process runs without errors. Many thanks.

**Comment by oMuransky** [comment, marked as answer]
The following process installs NEML, MOOSE, DEER, BLACKBEAR, and links them. Works for me on a clean WSL2/Ubunutu-20.04 (today).

#--------------- 
Update Ubuntu & Install pre-requisites  
#--------------- 
sudo apt update​  
sudo apt upgrade 
sudo apt install build-essential python3 libmpich-dev git python-is-python3 cmake m4 python3-dev​  
sudo apt install liblapack-dev libblas-dev libssl-dev bison flex 
cd ~​ 
#--------------- 
# Install required python packages 
#--------------- 
pip install packaging 
#--------------- 
# Install/update cmake 
#--------------- 
wget https://github.com/Kitware/CMake/releases/download/v3.23.0/cmake-3.23.0.tar.gz 
tar -zxvf cmake-3.23.0.tar.gz  
cd cmake-3.23.0/ 
./bootstrap 
make 
sudo make install 
#--------------- 
# Remove MPICH and install OPENMPI (not required)
#--------------- 
sudo apt-get purge --auto-remove mpich  
sudo apt install openmpi-bin 
#--------------- 
# Export paths in .bashrc file or .bash_profile
#--------------- 
export MOOSE_DIR=$MOOSE_DIR/home/path-to-moose/moose 
export NEML_DIR=$NEML_DIR/home/path-to-neml/neml 
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH/home/path-to-neml/neml 
source ~/.bashrc 
#--------------- 
# Install NEML 
#--------------- 
mkdir moose​ 
cd moose 
git clone https://github.com/Argonne-National-Laboratory/neml.git 
cd neml/ 
cmake . 
make -j 8
#--------------- 
# Install MOOSE 
#--------------- 
cd .. 
git clone https://github.com/idaholab/moose.git 
cd moose/ 
cd scripts/ 
./update_and_rebuild_petsc.sh --download-mumps=no --with-mumps=no 
./update_and_rebuild_libmesh.sh  
export MOOSE_DIR=/home/path-to-moose/moose 
cd ../framework/ 
make -j 8 
#--------------- 
# Install DEER 
#--------------- 
cd .. 
git clone https://github.com/Argonne-National-Laboratory/deer.git
cd deer/ 
make -j 8
#--------------- 
# Install BLACKBEAR 
#--------------- 
cd .. 
git clone https://github.com/idaholab/blackbear.git 
cd blackbear/ 
git submodule init 
git submodule update contrib/neml 
make -j 8

**Reply by milljm** [reply]
Do you think the swap between MPICH and OpenMPI was necessary? We've swapped back and forth over the years. This is more of a curiosity, than troubleshooting question. Cheers!

**Reply by oMuransky** [reply]
Hi Jason, we are trying to test the openmpi, because we do have some memory leak when running crystal plasticity simulations in DEER - it seems to be related to MPI, but I'm unsure as yet if replacing MPICH with OpenMPI will sort the problem. Cheers
